source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -795,10 +795,12 @@ namespace
             if (input)
             {
                 // Write input if it's connected or different from default value.
```

Comment:
Needed to modify this condition because the nodedef can have a uivisible of true (possibly from being non-existent) or false. If it's false then if the input has a uivisible of true then we need to write out the input. Or similar to the previous version if the nodedef uivisible is true and the input has a uivisible of false we also need to write things out.

Reply:
Sorry but butting in but isn't this just the input value is not equal to the definition value so you need to write it out. So can there be a generic function to handle this like is done in RtValue::compare(), perhaps via a an RtAttribute::compare() ?

Reply:
Maybe something like this to make it more readable:

```
const mx::RtTypedValue* uiVisible1 = input.getMetadata(UI_VISIBLE);
const mx::RtTypedValue* uiVisible2 = attrDef.getMetadata(UI_VISIBLE);
const bool uiHidden1 = uiVisible1 && (uiVisible1->getValueString() == VALUE_STRING_FALSE);
const bool uiHidden2 = uiVisible2 && (uiVisible2->getValueString() == VALUE_STRING_FALSE);
const bool writeUiHidden = uiHidden1 != uiHidden2;

if (writeDefaultValues || writeUiHidden .... )
{
```